### PR TITLE
Updated set of optional parameters for Yandex Metrica integration

### PR DESCRIPTION
**Pull request by mrhyde**
Hi guys, i decided to update this integration mostly because of need to set custom parameters for Yandex Metrica, and since there was a couple of other, missing, i added them too.

You can find official documentation with description of each parameter [here](http://help.yandex.com/metrica/objects/creating-object.xml)

P.S. I intentionally left `ut` parameter aside since it's pretty easy to mess up with it and default value is not clear (empty string for `no`, but `0` is the same as `noindex` looks pretty weird to me). And `defer` parameter will probably break your integration so there is no need of that unless someone will add support for it.

By the way, your service is amazing :+1:

### CODE REVIEW

**Comment by sperand-io** [comment]
Thanks @MrHyde 

ping :) @ndhoule

**Comment by mrhyde** [comment]
Hi guys, do you have any estimate on this? Is there anything i can do to help you?

**Comment by mrhyde** [comment]
This is extremely disappointing, I literally cant see any reasons to keep using segment when devs are not able to accept a simple PR, not to mention that project like this must guarantee that all integrations are up to date and function properly...

**Comment by mrhyde** [comment]
Bounce rate tracking in Yandex.Metrika before (with segment) and after (without)

File: lib/index.js [hunk 1]
```diff
@@ -16,30 +16,45 @@ var Yandex = module.exports = integration('Yandex Metrica')
...
   .global('yandex_metrika_callbacks')
   .global('Ya')
+  .option('type', 0)
   .option('counterId', null)
   .option('clickmap', false)
   .option('webvisor', false)
+  .option('trackHash', false)
+  .option('trackLinks', false)
+  .option('accurateTrackBounce', false)
+  .option('params', null)
```
Comment (f2prateek) [inline lib/index.js]: We'll need to take this out. We can't render arbitrary parameters through our UI.

**Comment by f2prateek** [comment]
LGTM, just one nit and we can update our app :)

**Comment by mrhyde** [comment]
@f2prateek thank you for reviewing this, we are looking forward to seeing it in your app :)

File: lib/index.js (hunk 1 above)
Comment (francisco) [inline lib/index.js]: @mrhyde Just to confirm - the above params options needs to be removed from the PR along with lines 46 and 57.

If you absolutely need to send the params, let us know and we can work on a solution for handling that.

File: lib/index.js [hunk 2]
```diff
@@ -16,30 +16,45 @@ var Yandex = module.exports = integration('Yandex Metrica')
   .assumesPageview()
   .global('yandex_metrika_callbacks')
   .global('Ya')
+  .option('type', 0)
   .option('counterId', null)
   .option('clickmap', false)
   .option('webvisor', false)
+  .option('trackHash', false)
+  .option('trackLinks', false)
+  .option('accurateTrackBounce', false)
```
Comment (f2prateek) [inline lib/index.js:25]: Also any reason why this is false by default? Seems to me you'd always want an accurate bounce rate.

Comment (f2prateek) [inline lib/index.js:25]: Actually should be fine since that's the yandex default as well https://cloudup.com/cW0cC7VA51q.

Comment (mrhyde) [inline lib/index.js:25]: I left it `false` for exact same reason - that's a default value, but enabling accurate bounce rate should only positively impact your metrics. That was the main goal of this PR

Comment (f2prateek) [inline lib/index.js:25]: :+1:

File: lib/index.js (hunk 1 above)
Comment (mrhyde) [inline lib/index.js]: @francisco Sure. Is it possible to do something like passing data from `analytics.track()` or `analytics.identify()` Yandex.Metrika afterwards via `params`? But I think that would just duplicate the visit amount :/

Comment (f2prateek) [inline lib/index.js]: The best way would be to define a global object outside of segment that contains the parameters you want to set. Then when we initialize yandex, we can use that global variable.

https://github.com/segment-integrations/analytics.js-integration-sentry/blob/master/lib/index.js#L47

Can you explain what kind of things you'd want to set via this params object?

Comment (mrhyde) [inline lib/index.js]: We are using it basically for two things
A/B testing https://yandex.com/support/metrica/reports/visit-params.xml
and tracking Goals specific parameters https://yandex.com/support/metrica/objects/reachgoal.xml#reachgoal

**Comment by ArmorDarks** [comment]
So... it stuck again?..

**Comment by f2prateek** [comment]
@ArmorDarks the PR still needs to remove the `params` option  https://github.com/segment-integrations/analytics.js-integration-yandex-metrica/pull/1#discussion_r43073978

**Comment by mrhyde** [comment]
This PR is going to celebrate its first birthday soon...

**Comment by imevro** [comment]
Guys, without this PR we have bounce rate **at 70%** and our positions in yandex search are very low. 

image

Meanwhile in Google Analytics we have bounce rate **lower than 20%**.
image

cc segment team @sperand-io @f2prateek @francisco @anthonyshort @calvinfo @dominicbarnes @ivolo @jpetey75 @lambtron @ndhoule @rakeshnair @reinpk @stephenmathieson @stevenmiller888 @tejasmanohar @thehydroimpulse @therebelrobot @tsholmes @yields

**Comment by ArmorDarks** [comment]
@evgenyrodionov Did you mean _without it_? It hasn't been merged yet.

**Comment by ArmorDarks** [comment]
Oh my f****, this deserves a cake! Thanks @hankim813

**Comment by sperand-io** [comment]
@evgenyrodionov @ArmorDarks @mrhyde our sincerest apologies for the delay here. I've added the appropriate settings metadata so as soon as this goes live this afternoon you'll see analytics.js rendering the new and improved initialization logic.

**Comment by hankim813** [comment]
deployed!

**Comment by ArmorDarks** [comment]
That's great, thanks guys!

**Comment by imevro** [comment]
Thanks, guys!

**Comment by ArmorDarks** [comment]
@sperand-io @hankim813  btw, will we get new options in UI for those newly added parameters?

**Comment by imevro** [comment]
@ArmorDarks already

image

**Pull request merged**